musicbot/bot.py

```diff
@@ -2741,6 +2741,11 @@ async def on_message(self, message):
                 handler_kwargs['guild'] = message.guild
 
             if params.pop('player', None):
+                #check if the bot is already in the channel when the command 'play' is executed
+                #otherwise, attempt auto summon
+                if message.guild.id not in self.players and command == "play":
+                    if message.author.id == self.config.owner_id or (user_permissions.command_whitelist and command in user_permissions.command_whitelist and user_permissions.command_blacklist and command not in user_permissions.command_blacklist):
```

Comment:
The same if statements are further down in the code here:

```py
            if message.author.id != self.config.owner_id:
                if user_permissions.command_whitelist and command not in user_permissions.command_whitelist:
                    raise exceptions.PermissionsError(
                        "This command is not enabled for your group ({}).".format(user_permissions.name),
                        expire_in=20)
                elif user_permissions.command_blacklist and command in user_permissions.command_blacklist:
                    raise exceptions.PermissionsError(
                        "This command is disabled for your group ({}).".format(user_permissions.name),
                        expire_in=20)
```

...so I'd probably recommend splitting that out to its own function and calling the function instead.